phantazm-api/src/main/java/com/github/phantazmnetwork/api/ClientBlockHandlerSource.java

```diff
@@ -0,0 +1,17 @@
+package com.github.phantazmnetwork.api;
+
+import net.minestom.server.instance.Instance;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Provides {@link ClientBlockHandler} instances corresponding to Minestom {@link Instance} objects.
+ */
+@FunctionalInterface
+public interface ClientBlockHandlerSource {
+    /**
+     * Creates or returns an existing {@link ClientBlockHandler} instance for the provided Minestom {@link Instance}.
+     * @param instance the instance for which to create or retrieve a ClientBlockHandler
+     * @return the ClientBlockHandler for the given instance
+     */
+    @NotNull ClientBlockHandler forInstance(@NotNull Instance instance);
```

Comment:
it might be useful to have per-player fakeblocks?

Reply:
would need a different interface to produce those, this one is just for instance-wide implementations

Reply:
unless you see a way to merge the two somehow

Reply:
this is ok